docs/adr/0011-operators.md

```diff
@@ -0,0 +1,145 @@
+
+## Decision Outcome
+
+Chosen option: _"3. Find balance between 1. and 2."_, because it is desired to keep track of all resources which may effect other users and other namespaces i.e. operators provisioned to manage all cluster namespaces.
+
+Proposed decision defines strict rules for given resources. These rules are subject to future updates if needed, but the repositories and applications are required to consistently enforce them:
+
+### Operator Lifecycle Manager compatible operators
+
+Operator Lifecycle Manager defines following resource kinds:
+
+- Cluster service version
+- Catalog source
+- Subscription
+- Install plan
+- Operator groups
+
+These are the set of guidelines we want to enforce on them:
+
+#### Cluster service version
+
+This resource is not meant to be deployed manually and is expected to be provided from a _Catalog source_. If there's really a need to deploy a custom _Cluster service version_, the resource has to live within the `cluster-scope` application.
+
+#### Catalog source
+
+It is uncommon for a _Catalog source_ to be declared manually. Operators are expected to be provisioned from the default catalogs and it is advised to rather publish the desired operator to a community catalog than deploying a custom _Catalog source_. If it is really inevitable to deploy an additional `CatalogSource` resource, this resource is required to be deployed to `openshift-marketplace` namespace and therefore it is required to declare this resource within the `cluster-scope` application.
+
+#### Subscription
+
+A `Subscription` resource is a namespace-scoped resource. It is effectively responsible for the operator deployment. Therefore we recognise 2 possible scenarios and use cases for this resource:
+
+1. A namespaced operator with a single namespace visibility:
+
+   The `Subscription` resource can be hosted along with application manifests in an application repository. This application has to be deployed to the same namespace as specified for the subscription.
+
+2. An operator with all namespaces visibility:
+
+   For operators with global effect, we expect the `Subscription` resource to be hosted in `cluster-scope` application only.
+
+#### Install plan
+
+It is not expected to be deploying a custom _Install plan_. We expect every plan to be provided by the _Cluster service version_ resource belonging to the given operator.
+
+#### Operator groups
+
+An `OperatorGroup` resource is a namespace-scoped resource. However we require to be place this resource in `cluster-scope` application only due to following reasons:
+
+1. This resource is a singleton in namespace context. There can't be multiple instances of the said resource in the same namespace, even if the name differs.
+2. When deployed it generates `ClusterRole` and `ClusterRoleBinding` and aggregation rules.
+3. It can specify a all namespaces visibility for an operator.
```

Comment:
A 4th point would be that by default project admins do not have create/edit rights to `operatorgroups` (due to points 2 and 3), so cluster-admins are required anyway to deploy/update operator groups.

Reply:
Rephrased as:
4. Dy default project admins do not have create/edit rights to `OperatorGroup`s (due to points 2 and 3), so only cluster-admins can manage _Operator groups_.